commands/mr/checkout/mr_checkout.go

```diff
@@ -37,20 +36,13 @@ func NewCmdCheckout(f *cmdutils.Factory) *cobra.Command {
 				return err
 			}
 
-			repo, err := f.BaseRepo()
+			mr, repo, err := mrutils.MRFromArgs(f, args)
 			if err != nil {
 				return err
 			}
 
-			mrID := utils.StringToInt(args[0])
-
-			mr, err := api.GetMR(apiClient, repo.FullName(), mrID, &gitlab.GetMergeRequestsOptions{})
-
-			if err != nil {
-				return err
-			}
 			if mr == nil {
-				return fmt.Errorf("merge Request !%d not found\n", mrID)
+				return fmt.Errorf("merge request from %q not found", args[0])
 			}
```

Comment:
This code will always be unreachable since the error is handled at line 40-42. 
If the `mr` returned by `mrutils.MRFromArgs` is nil, it is accompanied by an error.

Reply:
Thanks, fixed